### ISSUE TITLE
Poker Hand Prediction

### ISSUE THREAD

**Issue by aviralgarg05** [abhisheks008/ML-Crate#693]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Poker Hand Prediction
:red_circle: **Aim** : To Predict the Poker Hand
:red_circle: **Dataset** : https://archive.ics.uci.edu/dataset/158/poker+hand
:red_circle: **Approach** : 1. Data Pre-processing and classification 2. Visualization  3. Training and Modelling - DT, RF etc. 4. Metrics and Accuracy Scores

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name :  Aviral Garg
- GitHub Profile Link : https://github.com/aviralgarg05
- Participant ID (If not, then put NA) : NA
- Approach for this Project : 1. Data Pre-processing and classification 2. Visualization  3. Training and Modelling - DT, RF etc. 4. Metrics and Accuracy Scores
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.) SSOC Contributor

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
One issue at a time.

**Comment by siddhant4ds** [comment]
@abhisheks008 If this issue is still available, I would like to work on it. I have closed my previous issue.

Full name: Siddhant Tiwari
GitHub profile link: https://github.com/siddhant4ds
Participant ID: sid4ds (Devfolio), sid4ds (Discord)
Participant role: SSOC-3 Contributor
Approach:
1. Exploratory data analysis
2. Feature engineering based on domain knowledge of cards and poker rules.
3. Spot-checking some algorithms such as Logistic Regression and Decision Trees for baseline performance.
4. Focus on Gradient-boosting models (XGBoost, CatBoost, LightGBM), since the dataset is fairly large with categorical features, and the problem is multi-class classification.

**Comment by abhisheks008** [comment]
> @abhisheks008 If this issue is still available, I would like to work on it. I have closed my previous issue.
> 
> Full name: Siddhant Tiwari GitHub profile link: https://github.com/siddhant4ds Participant ID: sid4ds (Devfolio), sid4ds (Discord) Participant role: SSOC-3 Contributor Approach:
> 
> 1. Exploratory data analysis
> 2. Feature engineering based on domain knowledge of cards and poker rules.
> 3. Spot-checking some algorithms such as Logistic Regression and Decision Trees for baseline performance.
> 4. Focus on Gradient-boosting models (XGBoost, CatBoost, LightGBM), since the dataset is fairly large with categorical features, and the problem is multi-class classification.

This issue is opened by another contributor, hence can't be assigned to you.

**Comment by siddhant4ds** [comment]
No problem. Will look for another one. Thanks.

**Comment by aviralgarg05** [comment]
Hey can you assign me this issue now...

**Comment by abhisheks008** [comment]
Hi @aviralgarg05 implement 6-7 models for this problem statement. Assigned this issue to you.
